addons/web/controllers/main.py

```diff
@@ -506,8 +506,21 @@ def binary_content(xmlid=None, model='ir.attachment', id=None, field='datas', un
 
     # get content after cache control
     if model == 'ir.attachment' and obj.type == 'url' and obj.url:
```

Comment:
Should be done before at line 480 to handle headers correctly (mimetype should be guessed from filename)